src/ansys_sphinx_theme/__init__.py

```diff
@@ -538,6 +538,83 @@ def check_for_depreciated_theme_options(app: Sphinx):
         )
 
 
+def extract_whatsnew(app, doctree, docname):
+    """Extract the what's new content from the document."""
+    config_options = app.config.html_theme_options
+
+    whats_new_options = config_options.get("whatsnew")
+    if not whats_new_options:
+        return
+    no_of_contents = whats_new_options.get("no_of_headers", 3)
+    document_name = whats_new_options.get("file", "release-note")
+    get_doctree = app.env.get_doctree(document_name)
+    whats_new_content = []
+    docs_content = get_doctree.traverse(nodes.section)
+    app.env.whatsnew_content = []
+
+    if not docs_content:
+        return
+
+    versions_nodes = [node for node in docs_content if node.get("ids")[0].startswith("version")]
+
+    # get the version nodes upto the specified number of headers
+    versions_nodes = versions_nodes[:no_of_contents]
+
+    if not versions_nodes:
+        return
+
+    for version_node in versions_nodes:
+        title = version_node[0].astext()
+        sections = list(version_node.traverse(nodes.section))
+
+        whats_new_nodes = [node for node in sections if node[0].astext().lower() == "whatsnew"]
+
+        if not whats_new_nodes:
+            continue
+
+        children = [node for node in whats_new_nodes[0].traverse(nodes.section)]
+
+        headers = [child[0].astext() for child in children]
+
+        # header_anchors = [child.get("ids")[0] for child in children]
+
+        if len(children) > 1:
+            children = headers[1:]
+
+        else:
+            children = [whats_new_nodes[0].traverse(nodes.paragraph)[0].astext()]
+
+        contents = {
+            "title": title,
+            "title_url": f"{document_name}.html#{version_node.get('ids')[0]}",
+            "children": children,
+            "url": f"{document_name}.html#{whats_new_nodes[0]['ids'][0]}",
+        }
+
+        whats_new_content.append(contents)
+
+    app.env.whatsnew_content = whats_new_content
+
+
+def add_whatsnew(app, pagename, templatename, context, doctree):
+    """Add what's new section to the context."""
+    config_options = app.config.html_theme_options
+    whats_new_options = config_options.get("whatsnew")
+    if not whats_new_options:
+        return
+
+    pages = whats_new_options.get("pages", ["index"])
```

Comment:
Does this mean that any page named `index` will display the "What's new" section?

Reply:
No only the landing page will have the name `index` all other index pages will be `dir/index` for example `getting-started/index`

Reply:
Ah! Nice. Thanks for clarifying.